src/main/kotlin/org/jitsi/rtp/rtp/RtpPacket.kt

```diff
@@ -208,28 +269,33 @@ open class RtpPacket(
         // B: this.getHeaderLength()
         // C: this.getPayloadLength()
         // D: this.buffer.length - this.length - this.offset
-        // We will try to extend the packet so that it uses A and/or D if
+
+        // If the newly-encoded header extensions block results in a header
+        // longer than the current one,
+        // we will try to extend the packet so that it uses A and/or D if
         // possible, in order to avoid allocating new memory.
 
         // We get this early, before we modify the buffer.
+        val currHeaderLength = headerLength
         val currPayloadLength = payloadLength
-        val extensionBit = hasExtensions
-        val extHeaderOffset = RtpHeader.FIXED_HEADER_SIZE_BYTES + csrcCount * 4
-
-        // This is an upper bound on the required length for the packet after
-        // the addition of the new extension. It is easier to calculate than
-        // the exact number, and is relatively close (it may be off by a few
-        // bytes due to padding)
-        val maxRequiredLength = length +
-                (if (extensionBit) 0 else RtpHeader.EXT_HEADER_SIZE_BYTES) +
-                1 /* the 1-byte header of the extension element */ +
-                extDataLength +
-                3 /* padding */
+        val baseHeaderLength = RtpHeader.FIXED_HEADER_SIZE_BYTES + csrcCount * 4
+
+        val newExtHeaderLength = if (pendingHeaderExtensions.isEmpty()) {
+            0
+        } else {
+            val rawHeaderLength = RtpHeader.EXT_HEADER_SIZE_BYTES +
+                pendingHeaderExtensions.sumBy { h -> h.totalLengthBytes }
+            val paddingLength = (4 - (rawHeaderLength % 4)) % 4
```

Comment:
We have `RtpUtils.getNumPaddingBytes` which does the same (I don't know if the performance is different)